File: src/main/java/net/yigitak/todoapp/models/RecurrentTaskTemplate.java
package net.yigitak.todoapp.models;


import jakarta.validation.Valid;
import jakarta.validation.constraints.NotEmpty;
import lombok.*;

import java.util.List;


@NoArgsConstructor
@Getter
@Setter
@Builder
@AllArgsConstructor
public class RecurrentTaskTemplate {

  @NotEmpty( message = "Task title cannot be empty." )
  String title;

  @Valid
  List< RecurrentSubtaskTemplate > subtasks;

  String description;

}

File: src/main/java/net/yigitak/todoapp/models/RecurrentTaskTemplate.kt
package net.yigitak.todoapp.models

import jakarta.validation.Valid
import jakarta.validation.constraints.NotBlank

data class RecurrentTaskTemplate(

    @field:NotBlank(message = "Task title cannot be blank.")
    val title: String,

    @field:Valid
    val subtasks: List<RecurrentSubtaskTemplate>,

    val description: String? = null
)
